lib/matplotlib/tight_layout.py

```diff
@@ -123,6 +123,9 @@ def auto_adjust_subplotpars(fig, renderer,
     for subplots, ax_bbox, (num1, num2) in zip(subplot_list,
                                                ax_bbox_list,
                                                num1num2_list):
+        if all([not ax.get_visible() for ax in subplots]):
```

Comment:
shouldn't this just be filtering out invisible axes?  `subplots = [ax for ox in subplots if ax.get_visible()]`

Would it make more sense to make sure what ever _calls_ method does that filtering?